grep-0001-coding-guidelines.md

```diff
@@ -73,6 +72,20 @@ and follow them.
   can thus be optimized better at compile time. In many cases, lambdas are
   also more readable than the `bind` calls, which have add their own
   syntactical idiosyncrasies (such as the `_1` argument etc.).
+* Feel free to use modern C/C++ features even if they were not used before.
+  Make sure they work with the compilers and dependencies which are set for the
+  version of GNU Radio the commit will be made upon. The GNU Radio CI system
+  will be able to confirm this. (Note: C++11 features are available starting
+  with GNU Radio version 3.8).
+
+#### Code Style Guidelines
+* Use the [LLVM code formatting guide](https://llvm.org/docs/CodingStandards.html#source-code-formatting). Following points will specify differences to it and highlight important points.
+* Do not indent namespaces
+* Use 4 spaces indentation width for new code.
+* Use spaces instead of tabs. If extending an existing file stick with the given indentation level. Do _not_ reindent a whole file and indent code consistently.
+* Before checking in code use tools in the code tree to sanitze your patches
```

Comment:
Typo (sanitize). Also, do we want to recommend tools? I think we should.

Reply:
I'm working on a second GREP discussing about how we want to implement changes to formatting going forward.

Reply:
Looking forward to that! It'll be a great help for you contributors.

Reply:
...and we should probably do a huge formatting patch on both maint-3.7 and master to reset it once and for all.